packages/runtime/runtime-utils/src/summarizerNode/summarizerNode.ts

```diff
@@ -128,14 +159,16 @@ export class SummarizerNode implements IRootSummarizerNode {
 		trackState: boolean = true,
 		telemetryContext?: ITelemetryContext,
 	): Promise<ISummarizeResult> {
-		assert(
-			this.isSummaryInProgress(),
-			0x1a1 /* "summarize should not be called when not tracking the summary" */,
-		);
-		assert(
-			this.wipSummaryLogger !== undefined,
-			0x1a2 /* "wipSummaryLogger should have been set in startSummary or ctor" */,
-		);
+		if (trackState) {
+			assert(
+				this.isSummaryInProgress(),
+				0x1a1 /* "summarize should not be called when not tracking the summary" */,
+			);
+			assert(
+				this.wipSummaryLogger !== undefined,
+				0x1a2 /* "wipSummaryLogger should have been set in startSummary or ctor" */,
+			);
+		}
 
 		// Try to reuse the tree if unchanged
 		if (this.canReuseHandle && !fullTree && !this.hasChanged()) {
```

Comment:
You need to check for trackState here. If trackState is false, it should not enter this block.

Reply:
On a separate note, we should probably remove `trackState` entirely. It is redundant with `fullTree`. The only problem I see is tests which call summarize directly.